lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts

```diff
@@ -398,106 +237,126 @@ describe('AttachFileCloudWidgetComponent', () => {
             expect(widget.rootNodeId).toEqual('mock-node-id');
         });
 
-        it('Should be able to set given alias as rootNodeId if the nodeId of the alias is not fetched from the api', async () => {
-            widget.field = new FormFieldModel(new FormModel(), {
-                type: FormFieldTypes.UPLOAD,
-                value: []
-            });
-            widget.field.id = 'attach-file-alfresco';
-            widget.field.params = <FormFieldMetadata> allSourceWithRootParams;
+        it('should be able to use mapped string variable value if the destinationFolderPath set to string type variable', async () => {
+            const fetchNodeIdFromRelativePathSpy = spyOn(contentCloudNodeSelectorService, 'fetchNodeIdFromRelativePath').and.returnValue(mockNodeIdBasedOnStringVariableValue);
+
+            const form = new FormModel({ formVariables, processVariables});
+            createUploadWidgetField(form, 'attach-file-alfresco', [], mockAllFileSourceWithStringVariablePathType);
+            fixture.detectChanges();
+            await fixture.whenStable();
+            clickOnAttachButton('attach-file-alfresco');
             fixture.detectChanges();
             await fixture.whenStable();
-            const attachButton: HTMLButtonElement = element.querySelector('#attach-file-alfresco');
+            fixture.detectChanges();
 
-            expect(attachButton).not.toBeNull();
+            const alias = '-root-';
+            const opt = { relativePath: '/pathBasedOnStringvariablevalue' };
+            expect(fetchNodeIdFromRelativePathSpy).toHaveBeenCalledWith(alias, opt);
```

Comment:
let's leave always a space between the test's operations and the expectations. Please fix all the occurrences

Reply:
Updated, Thanks